resources/update.rb

```diff
@@ -83,7 +83,7 @@ def updates
       # Searches non installed updates
       search_result = session.CreateUpdateSearcher.Search 'IsInstalled=0'
       # Transforms to ruby array for future use
-      search_result.Updates.each { |update| updates << update }
+      search_result.Updates.each { |update| updates << update } unless search_result.nil?
```

Comment:
Maybe you should raise an error when the search result is nil; this shouldn't happen and could lead to invalid convergence (i.e. the node converge even if the updates have not been installed)

Reply:
Yet we are sometimes getting nil for Updates.  The search result may contain helpful information: https://docs.microsoft.com/en-us/windows/desktop/api/wuapi/nn-wuapi-isearchresult

Reply:
This is the issue, the `search_result` variable is *nil*; so we won't be able to get helpful info here.
The `IUpdateSearcher.Search` method should not return *nil* (https://docs.microsoft.com/en-us/windows/desktop/api/wuapi/nf-wuapi-iupdatesearcher-search)

Do you have some data about the context where/when this error occurred?